hnn_core/network.py

```diff
@@ -776,27 +817,50 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
         # This will be updated depending on the number of target cells
         # of each cell type
         src_idx = 0
-        for target_cell_type in target_populations:
+
+        # seed_increment increased by 1 for each target cell type,
+        # added to conn_seed to ensure statistical independence of random
+        # connections when probability < 1.0
+        for seed_increment, target_cell_type in enumerate(target_populations):
             target_gids = list(self.gid_ranges[target_cell_type])
             delays = delays_by_type[target_cell_type]
+            probability = probability_by_type[target_cell_type]
             if cell_specific:
                 target_gids_nested = [[target_gid] for
                                       target_gid in target_gids]
                 src_idx_end = src_idx + len(target_gids)
                 src_gids = (list(self.gid_ranges[name])
                             [src_idx:src_idx_end])
                 src_idx = src_idx_end
-                for receptor in weights_by_type[target_cell_type]:
+                for receptor_idx, receptor in enumerate(
+                        weights_by_type[target_cell_type]):
                     weights = weights_by_type[target_cell_type][receptor]
-                    self.add_connection(src_gids, target_gids_nested,
-                                        location, receptor, weights, delays,
-                                        space_constant)
+                    self.add_connection(
+                        src_gids=src_gids, target_gids=target_gids_nested,
+                        loc=location, receptor=receptor, weight=weights,
+                        delay=delays, lamtha=space_constant,
+                        probability=probability,
+                        conn_seed=drive['conn_seed'] + seed_increment)
+                    # Ensure that AMPA/NMDA connections target the same gids
+                    if receptor_idx > 0:
+                        self.connectivity[-1]['src_gids'] = \
+                            self.connectivity[-2]['src_gids']
```

Comment:
Is this necessary given that `conn_seed` applies the same `seed_increment` for each receptor of a given cell type?

Reply:
This is actually the root of the problem we were discussing with respect to enforcing matching connections. While technically these would share the same connections when providing the same seed, it's not very transparent, so they are explicitly set equal here.

Reply:
Hmm, I think the root problem is addressed by explicitly implementing (with documentation) that random connections are assigned on a cell-by-cell basis rather than a synapse-by-synapse basis. This is perfectly implemented by setting the seed increment for each cell type in an outer loop and then establishing each receptor's connection in an inner loop. That said, maybe the explicit check is good for maintaining this established convention in the future....